### PR TITLE
Initializes tbds Ruby Gem

### PR DESCRIPTION
**Pull request by dimanyc**
- initializes `tbds (0.0.1)` gem
- adds `tbds_avatar` view helper

to test `tbds_avatar` in a new Rails application: 

1. spin up a new rails app:
navigate to where you want to new rails app to be installed, and type the following in your terminal:
```bash
rails new [name_of_your_app]
```
2. In the Gemfile of your new app add:
```ruby
gem "tbds", git: "https://github.com/thoughtbot/design-system.git", branch: "dn/tbds-gem_v2"
```
3. navigate to the root of your new rails application in the terminal, and type:
```bash
rails g controller home index
```
4. in `routes.rb`, add the following
```ruby
root to: "home#index"
```
5. in `app/assets/stylesheets/home.scss`, add the following:
```scss
@import "tbds";
```
6. in `app/views/home/index.html.erb` add:
```erb
<%= tbds_avatar(
  path: "https://placekitten.com/g/200/300",
  size: :large,
  shape: :circle,
  alt_text: "design-system"
) %>

<%= tbds_avatar(
  path: "https://placekitten.com/g/200/300",
  size: :small,
  shape: nil,
  alt_text: "design-system"
) %>

<%= tbds_avatar(
  path: "https://placekitten.com/g/200/300",
  size: :large,
  shape: nil,
  alt_text: "design-system"
)%>

```
7. Start your application with `rails s` and navigate to `http://localhost:3000` in the browser

### CODE REVIEW

File: src/_tbds.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+@import "index";
```
Comment (tysongach) [inline src/_tbds.scss]: This file effectively duplicates what `index.scss` is doing (imports the system), and I think we should combine them. This file is also a Sass manifest, not a partial, and so it should not have an underscore prefix in the filename.

So, we should…

1. Delete this file.
1. Rename `src/index.scss` to `src/tbds.scss`
1. Update the path in the [installation instructions in the readme](https://github.com/thoughtbot/design-system#installation-for-ruby-on-rails):
    ```diff
    -@import "@thoughtbot/design-system/src/index";
    +@import "@thoughtbot/design-system/src/tbds";
    ```

File: bin/tbds* [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env ruby
```
Comment (tysongach) [inline bin/tbds*]: @dimanyc What does the `*` in this filename represent? I don’t think I’ve seen that before and I’m curious.

Comment (tysongach) [reply bin/tbds*]: From @dimanyc - “that's a vim thing showing files with specific permission modifiers. I think we can drop the * here”

**Comment by tysongach** [comment]
@dimanyc I just tested the gem by adding it to Hub, locally. I ran into two issues:

1. `size` and `shape` appear to be required. Can we make them optional?
1. We need a way to pass into extra CSS classes to be added to the `` (in addition to the tbds classes)

**Comment by dimanyc** [comment]
> @dimanyc I just tested the gem by adding it to Hub, locally. I ran into two issues:
> 
> 1. `size` and `shape` appear to be required. Can we make them optional?
> 2. We need a way to pass into extra CSS classes to be added to the `` (in addition to the tbds classes)

yes to both questions. Will have it done this Friday.

**Comment by tysongach** [comment]
@dimanyc You are amazing.

**Comment by dimanyc** [comment]
@tysongach try now

**Comment by tysongach** [comment]
@dimanyc - I can’t seem to get the `class` argument to work for me. Here is what I was trying:

```erb
<%= tbds_avatar(
  path: session_person.image_url,
  alt_text: "Your profile",
  class: "nav-avatar-image",
) %>
```

The output is (note that `nav-avatar-image` is missing from `class`):

```html
<img alt="Your profile" class="tbds-avatar " src="https://images.thoughtbot.com/people/images/web_medium/tyson-gach.jpg?1496859823">
```

**Comment by dimanyc** [comment]
> @dimanyc - I can’t seem to get the `class` argument to work for me. Here is what I was trying:
> 
> ```
> <%= tbds_avatar(
>   path: session_person.image_url,
>   alt_text: "Your profile",
>   class: "nav-avatar-image",
> ) %>
> ```
> 
> The output is (note that `nav-avatar-image` is missing from `class`):
> 
> ```
> Your profile
> ```

whoops. Forgot to add `class: args[:class]` to `AvatarPresenter`. Fixed!

**Comment by LkeMitchll** [comment]
@dimanyc Apologies this closed when the mater branch was deleted, could you rebate against `main`, and either reopen or create a new PR?

**Comment by dimanyc** [comment]
Thanks for the heads up @LkeMitchll! I think this repo is not being worked on anymore. I'm fine with keeping this closed for now until we decide to revive it.

**Comment by dimanyc** [comment]
rebased and re-opened here: https://github.com/thoughtbot/design-system/pull/200

**Pull request closed without merging**